src/Components/Blazor/WebAssembly.Authentication/test/WebAssemblyAuthenticationServiceCollectionExtensionsTests.cs

```diff
@@ -0,0 +1,256 @@
+            Assert.Equal("https://www.example.com/base/authentication/logout-callback", provider.PostLogoutRedirectUri);
+        }
+
+        [Fact]
+        public void OidcOptions_DefaultsCanBeOverriden()
+        {
+            var builder = WebAssemblyHostBuilder.CreateDefault();
+            builder.Services.AddOidcAuthentication(options =>
+            {
+                options.AuthenticationPaths = new RemoteAuthenticationApplicationPathsOptions
+                {
+                    LoginPath = "a",
+                    LoginCallbackPath = "a",
+                    LoginFailedPath = "a",
+                    RegisterPath = "a",
+                    ProfilePath = "a",
+                    RemoteRegisterPath = "a",
+                    RemoteProfilePath = "a",
+                    LogoutPath = "a",
+                    LogoutCallbackPath = "a",
+                    LogoutFailedPath = "a",
+                    LogoutSucceededPath = "a",
+                };
+                options.UserOptions = new RemoteAuthenticationUserOptions
+                {
+                    AuthenticationType = "a",
+                    ScopeClaim = "a",
+                    RoleClaim = "a",
+                    NameClaim = "a",
+                };
+                options.ProviderOptions = new OidcProviderOptions
+                {
+                    Authority = "a",
+                    ClientId = "a",
+                    DefaultScopes = Array.Empty<string>(),
+                    RedirectUri = "https://www.example.com/base/custom-login",
+                    PostLogoutRedirectUri = "https://www.example.com/base/custom-logout",
+                };
+            });
+
+            var host = builder.Build();
+
+            var options = host.Services.GetRequiredService<IOptions<RemoteAuthenticationOptions<OidcProviderOptions>>>();
+
+            var paths = options.Value.AuthenticationPaths;
+
+            Assert.Equal("a", paths.LoginPath);
+            Assert.Equal("a", paths.LoginCallbackPath);
+            Assert.Equal("a", paths.LoginFailedPath);
+            Assert.Equal("a", paths.RegisterPath);
+            Assert.Equal("a", paths.ProfilePath);
+            Assert.Equal("a", paths.RemoteRegisterPath);
+            Assert.Equal("a", paths.RemoteProfilePath);
+            Assert.Equal("a", paths.LogoutPath);
+            Assert.Equal("a", paths.LogoutCallbackPath);
+            Assert.Equal("a", paths.LogoutFailedPath);
+            Assert.Equal("a", paths.LogoutSucceededPath);
+
+            var user = options.Value.UserOptions;
+            Assert.Equal("a", user.AuthenticationType);
+            Assert.Equal("a", user.ScopeClaim);
+            Assert.Equal("a", user.RoleClaim);
+            Assert.Equal("a", user.NameClaim);
+
+            var provider = options.Value.ProviderOptions;
+            Assert.Equal("a", provider.Authority);
+            Assert.Equal("a", provider.ClientId);
```

Comment:
Also suggest giving different values to each of these

Reply:
done